include/pmacc/lockstep/ForEach.hpp

```diff
@@ -120,26 +128,31 @@ namespace pmacc
              */
             template<
                 typename T_Functor,
-                // check if functor is invocable
-                typename = Void<decltype(alpaka::core::declval<T_Functor>()(alpaka::core::declval<Idx const>()))>,
+                // check if functor is invokable
+                typename = IsCallableWithIndex<T_Functor>,
                 std::
                     enable_if_t<std::is_void<InvokeResult_t<T_Functor, Idx const>>::value && domainSize != 1, int> = 0>
             HDINLINE void operator()(T_Functor&& functor) const
             {
-                for(uint32_t i = 0u; i < numCollIter; ++i)
+                static constexpr uint32_t numFullLoops = domainSize / (simdSize * numWorkers);
+                for(uint32_t i = 0u; numFullLoops != 0u && i < numFullLoops; ++i)
```

Comment:
I have a few questions/suggestions here:

- Maybe add a comment above that we split loop into peeled (or we can call it full like currently) and remainder, and explain what peeled means?
- Isn't `numFullLoops != 0u` redundant? If it is 0, then `i < numFullLoops` would be false initially.
- The name `numFullLoops` is misleading, I would consider renaming to `numFullIterations` or `numFullLoopIterations`, or with `Peeled` instead of `Full`

Reply:
> Isn't numFullLoops != 0u redundant? If it is 0, then i < numFullLoops would be false initially.

`numFullLoops` is unsigned and therefore the compiler will throw warnings if `numFullLoops == 0`.
I added a comment.